aminator/plugins/cloud/ec2.py

```diff
@@ -306,36 +307,72 @@ def is_stale_attachment(self, dev, prefix):
 
     @registration_retry(tries=3, delay=1, backoff=1)
     def _register_image(self, **ami_metadata):
-        context = self._config.context
-        ami = Image(connection=self._connection)
-        ami.id = self._connection.register_image(**ami_metadata)
-        if ami.id is None:
+        """Register the AMI using boto3/botocore components which supports ENA
+           This is the only use of boto3 in aminator currently"""
+
+        # construct AMI registration payload boto3 style
+        request = {}
+        request['Name'] = ami_metadata.pop('name')
+        request['Description'] = ami_metadata.pop('description')
+        request['Architecture'] = ami_metadata.pop('architecture')
+        request['BlockDeviceMappings'] = ami_metadata.pop('block_device_map')
+        request['VirtualizationType'] = ami_metadata.pop('virtualization_type')
+        request['RootDeviceName'] = ami_metadata.pop('root_device_name')
+
+        if (ami_metadata.get('virtualization_type') == 'pv'):
```

Comment:
Similar to above, if `virtualization_type` or these others aren't in the dict, `ami_metadata.get()` is going to throw an exception.
May want to guard against this using something like `ami_metadata.get('some_key', None)` and doing a `is not None` check for those cases where the key may not exist and you want to avoid using the `None` value.